src/mumble/Log.cpp

```diff
@@ -812,9 +816,17 @@ void Log::log(MsgType mt, const QString &console, const QString &terse, bool own
 		}
 		if ((flags & Settings::LogSoundfile)
 			&& !(flags & Settings::LogMessageLimit && connectedUsers > Global::get().s.iMessageLimitUserThreshold)) {
-			QString sSound    = Global::get().s.qmMessageSounds.value(mt);
+			QString sSound = Global::get().s.qmMessageSounds.value(mt);
+
+			bool failedSound  = true;
 			AudioOutputPtr ao = Global::get().ao;
-			if (!ao || !ao->playSample(sSound, Global::get().s.notificationVolume)) {
+
+			if (ao) {
+				AudioOutputToken sample = ao->playSample(sSound, Global::get().s.notificationVolume);
+				failedSound             = (sample == AudioOutputToken::None());
+			}
+
+			if (failedSound) {
```

Comment:
This can also keep its original structure by making use of the implicit bool conversion